### ISSUE TITLE
macOS 10.13 release binary does not work

### ISSUE THREAD

**Issue by selsta** [monero-project/kovri#696]
If I don’t install boost with homebrew, I get the following error

```
selsta@mbpR ~/D/kovri-a8e0489-Darwin-x86_64-2017.08.28> ~/bin/kovri
dyld: Library not loaded: /usr/local/opt/boost/lib/libboost_system-mt.dylib
  Referenced from: /Users/selsta/bin/kovri
  Reason: image not found
fish: '~/bin/kovri' terminated by signal SIGABRT (Abort)
```

From the IRC
```
4:11 AM <i2p-relay> {-kovri-osx-10-12} selsta pigeons: interestingly, we appear to be successfully linking every dependency statically:
4:11 AM <i2p-relay> {-kovri-osx-10-12}         /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.4.0)
4:11 AM <i2p-relay> {-kovri-osx-10-12} $ otool -L kovri
4:11 AM <i2p-relay> {-kovri-osx-10-12} kovri:
4:11 AM <i2p-relay> {-kovri-osx-10-12}         /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
4:12 AM <i2p-relay> {-anonimal} selsta: so I wonder if this is 10.13 specific (though I don't see the obvious cause off the top of my head)
```

**Comment by selsta** [comment]
when using `make release-static` on 10.13 I get a working binary.

```
selsta@mbpR ~/d/k/build> otool -L kovri
kovri:
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

In comparison to the current nightly:

```
selsta@mbpR ~/bin> otool -L kovri
kovri:
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
	/usr/local/opt/boost/lib/libboost_system-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_thread-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_chrono-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_date_time-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_atomic-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/openssl/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
```

**Comment by anonimal** [comment]
If 10.13 static builds are backwards-compatible then https://github.com/monero-project/meta/issues/153 could be a fix. Otherwise, I'd need 10.13 access somewhere in order to debug/test.

**Comment by anonimal** [comment]
Collaborating with @selsta, building a `make release-static` binary on 10.13 and running otool on the static binary on both 10.11 and 10.12 (identical binary, different systems):

```
    OSX 10.11
     
    ./build/kovri:
            /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
            /Users/selsta/dev/kovri/deps/cpp-netlib/build/libs/network/src/libcppnetlib-client-connections.0.dylib (compatibility version 0.0.0, current version 0.12.0)
            /usr/local/opt/openssl/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
            /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
            /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
            /Users/selsta/dev/kovri/deps/cpp-netlib/build/libs/network/src/libcppnetlib-server-parsers.0.dylib (compatibility version 0.0.0, current version 0.12.0)
            /Users/selsta/dev/kovri/deps/cpp-netlib/build/libs/network/src/libcppnetlib-uri.0.dylib (compatibility version 0.0.0, current version 0.12.0) 
```

```
    OSX 10.12
     
    ./build/kovri:
            /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
            /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0) 
```

**Comment by anonimal** [comment]
The nightly build needs to move from building on 10.11 to 10.13.

On current master, `make release-static` on 10.13 now produces desired results on 10.11 and 10.12:

```
kovri:
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

Referencing https://github.com/monero-project/meta/issues/211.

**Comment by anonimal** [comment]
This was resolved in https://github.com/monero-project/meta/issues/211. There is an unrelated documentation build issue (https://build.getmonero.org/builders/kovri-static-osx/builds/467) but, once it's resolved, the nightly 10.13 will also work on 10.11 and 10.12.
